R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,262 @@
+      L123.in_EJ_CHINA_ownuse
+
+    L126.out_EJ_R_elecownuse_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.out_EJ_CHINA_ownuse
+
+    L123.in_EJ_CHINA_ownuse %>%
+      left_join_error_no_match(L123.out_EJ_CHINA_ownuse, by = c("GCAM_region_ID", "sector", "fuel", "year")) %>%
+      mutate(value = value.x - value.y) %>%
+      select(-value.x, -value.y) ->
+      L123.net_EJ_CHINA_ownuse
+
+    # Then build table with each province's share of the national ownuse. Note that this is assumed invariant over time.
+    L123.out_EJ_province_elec_F %>%
+      group_by(province, year) %>%
+      summarise(value = sum(elec_F)) %>%
+      ungroup %>%
+      group_by(year) %>%
+      mutate(value = value/sum(value)) %>%
+      ungroup  %>%
+      mutate(sector = 'electricity ownuse',
+              fuel = 'electricity') %>%
+      replace_na(list(value = 0)) ->
+      L123.net_pct_province_CHINA_ownuse_elec
+
+    # Net own use = national total by each province's share
+    L123.net_EJ_CHINA_ownuse %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.net_EJ_province_ownuse_elec
+
+    L123.net_pct_province_CHINA_ownuse_elec %>%
+      left_join_error_no_match(L123.net_EJ_province_ownuse_elec, by = c("sector", "fuel", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(-value.x, -value.y) ->
+      L123.net_EJ_province_ownuse_elec
+
+    # The input of the electricity_net_ownuse sector is equal to sum of all generation (industrial CHP + electric sector)
+    L123.out_EJ_province_elecind_F <- L123.out_EJ_province_elec_F
+    names(L123.out_EJ_province_elecind_F)[5] <- 'value'
+    L123.out_EJ_province_elecind_F %>%
+      bind_rows(L132.out_EJ_province_indchp_F) ->
+      L123.out_EJ_province_elecind_F
+
+
+    L123.out_EJ_province_elecind_F %>%
+      group_by(province, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      mutate(sector = 'electricity ownuse',
+             fuel = 'electricity') ->
+      L123.in_EJ_province_ownuse_elec
+
+
+    # Output of electricity_net_ownuse sector is equal to input minus ownuse "net" energy
+    L123.in_EJ_province_ownuse_elec %>%
+      left_join_error_no_match(L123.net_EJ_province_ownuse_elec, by = c('province', 'year')) %>%
+      mutate(value = value.x - value.y,
+             sector = sector.x,
+             fuel = fuel.x) %>%
+      subset(select = c(province, sector, fuel, year, value)) ->
+      L123.out_EJ_province_ownuse_elec
+
+    # ===================================================
+    L123.in_EJ_province_elec_F %>%
+      add_title("Electricity sector energy consumption by province and fuel") %>%
+      add_units("EJ") %>%
+      add_comments("The energy consumption is generated by dividing data on capacity factors by province by national average capacity factor") %>%
+      add_legacy_name("L123.in_EJ_province_elec_F") %>%
+      add_precursors("gcam-china/province_names_mappings",
```

Comment:
Please figure out which input files are actually used for each output when specifying precursors.